### ISSUE TITLE
RFE: SCT should provision cloud images in parallel

### ISSUE THREAD

**Issue by mykaul** [scylladb/scylla-cluster-tests#7025]
Example:
```
Dec 21 22:11:53.074455 ip-10-0-0-21 systemd[1]: Finished Scylla Cloud Image Setup service.
Dec 21 22:36:58.180713 cluster-base-scale-test-2024-1-db-node-de3ed443-4 systemd[1]: Starting Scylla Server...
```

The node was done with the cloud image and spent 25 minutes waiting for other nodes cloud images to be set up, as it is set up serially, I believe (with syslog-ng install, etc.)

(Log taken from https://argus.scylladb.com/test/9814ff28-d805-40fb-8bea-7c72bec26c3c/runs?additionalRuns%5B%5D=de3ed443-c506-4e69-badb-db12e481367b which does set up a lot of nodes!)

**Comment by fruch** [comment]
same as in #6973, nothing change since then

**Comment by fruch** [comment]
closing in favor of issue:
https://github.com/scylladb/scylla-cluster-tests/issues/7242
